### PR TITLE
[Kotlin] add apiSuffix configuration

### PR DESCRIPTION
**Pull request by fj-roman**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@bbdouglas  @sreeshas  @jfiala  @lukoyanov  @cbornet  @jeff9finger @karismann @Zomzog

### Description of the PR

Add an option to configure a suffix of generated API classes. Default suffix is still **API**.

### CODE REVIEW

**Comment by fj-roman** [comment]
Can someone please give me a clue about the issue with ci/circleci ?

**Comment by Zomzog** [comment]
For the last build, it looks like a CircleCi cache issue may be a core member can trigger a build without cache.

But for the first error, you must regenerate all impacted sample (R, c#, go...)

**Comment by karismann** [comment]
it seem like there's a regression with your change when apisuffix is not set :  for example FakeApi become Fake. Thats'why you have a lot of error with sample regenerate checking shell in Cirecle/CI.
Please try to rerun one of the sample locally (like java-petstore-jersey2.sh) to try to debug it

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -71,7 +71,7 @@
     protected Set<String> reservedWords = new HashSet<String>();
     protected Set<String> languageSpecificPrimitives = new HashSet<String>();
     protected Map<String, String> importMapping = new HashMap<String, String>();
-    protected String modelPackage = "", apiPackage = "", fileSuffix;
+    protected String modelPackage = "", apiPackage = "", apiSuffix = "", fileSuffix;
```
Comment (karismann) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: you forgot to define the default value of apiSuffix : 
apiSuffix = "Api"
that should resolve your CIRCLE/CI errors

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 2]
```diff
@@ -1575,7 +1583,7 @@ public String toApiName(String name) {
         if (name.length() == 0) {
             return "DefaultApi";
         }
-        return camelize(name) + "Api";
+        return (this.apiSuffix.isEmpty() ? camelize(name) : camelize(name) + this.apiSuffix);
```
Comment (karismann) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: that would be great to remove the `toApiName` overriding method in `AbstractJavaCodegen` class to be able to have this feature in most of java generators

**Comment by fj-roman** [comment]
Thanks for the help and code review!

**Comment by wing328** [comment]
@fj-roman thanks for adding the functionality 👍 

@karismann thanks for the review.

From what I understand, apiSuffix is added as a global option but only the Java generators support this option so far. Instead of making it a global option from day one, what about adding the option only to the Java generators (Abstract Java class) to start with? 

In other words, `apiSuffix` should be introduced as a generator-specified option.

**Comment by fj-roman** [comment]
@wing328 fair point. My main focus is to provide this feature for the KotlinSpringServerCodegen. I will refactor the the PR to move this feature to the AbstractKotlinCodegen. Maybe I can provide it to all Kotlin generators within one PR.

Should I close this PR and open later a new one, or should I just edit the description after I complete the refactoring?

**Comment by wing328** [comment]
@fj-roman why closed the PR?

**Comment by wing328** [comment]
Tested locally and it works as expected 👍

**Pull request merged**